api/v1alpha1/warehouse_types.go

```diff
@@ -55,6 +55,14 @@ type WarehouseSpec struct {
 	// empty, the defaulting webhook will set the value of this field to the value
 	// of the shard label.
 	Shard string `json:"shard,omitempty" protobuf:"bytes,2,opt,name=shard"`
+	// Interval is the interval at which the Warehouse should discover new
+	// artifacts. This field is optional. When left unspecified, the field is
+	// implicitly treated as if its value were "5m0s".
+	//
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(s|m|h))+$"
```

Comment:
This is a common workaround for the issue described here: https://github.com/kubernetes/apiextensions-apiserver/issues/56

I took the further liberty to drop `ms` from the regex, as we probably do not want people to go below the seconds (and the level of precision in combination with e.g. minutes has little value).